### ISSUE TITLE
stderr of the which command shows during Puppet runs on agent nodes

### ISSUE THREAD

**Issue by natemccurdy** [acidprime/r10k#241]
After the commit that added in a custom fact for `r10k_path` (1780577ba320714d863dc5a1f2a5bf49d968415e), my agent nodes (non-masters) are showing the text below during Puppet runs.
``` shell
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts 
/usr/bin/which: no r10k in (/usr/local/sbin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin:/root/bin)
...
```

What's happening is that `which` can't find r10k on my agents because r10k isn't installed on them. The stderr of `which` gets sent to the output of the Puppet run and totally ruins my day.

The `r10k_path` custom fact should catch stderr and hide it. Even better would be to not shell out to `which` and instead use native ruby to do this.

Maybe do something like http://stackoverflow.com/a/6624392

**Comment by acidprime** [comment]
fixed in f9e253df62a955937a93c50671a44bb262393eb8

**Comment by acidprime** [comment]
Released in `v3.1.1`
